File: scripts/load_tsdf_metadata_from_path.m
function [mat_list_metadata, mat_list_data] = load_tsdf_metadata_from_path(mat_metadata_path)
    % Load metadata and data from a path to a TSDF metadata json file
    % Input:
    %   mat_metadata_path: path to the .mat file containing the metadata
    % Output:
    %   mat_list_metadata: list of the metadata objects describing each binary file
    %   mat_list_data: list of time series data (each element representing a content of a binary file)

    % Add the path to the matlab scripts
    addpath(genpath('.'));
    
    % Convert the path to a string
    py_metadata_path = py.str(mat_metadata_path);

    % Run the python script load_tsdf_metadata_from_path.py and get the 3 outputs
    [py_succ, py_list_metadata, py_list_data] = pyrunfile("load_tsdf_metadata_from_path.py", ["Success" "Return_list_metadata" "Return_list_data"], path_to_metadata=py_metadata_path);

    % Convert Python boolean to MATLAB logical
    mat_succ = logical(py_succ);
    


    % Convert to MATLAB data structures in case the executions was successful 
    if mat_succ
      mat_list_metadata = py2mat_listmeta(py_list_metadata);
      mat_list_data = py2mat_listnumpy2(py_list_data);
    else
       mat_list_metadata = [];
       mat_list_data = [];
    end
    
    
end

File: scripts/tsdf_python_interface/load_tsdf_metadata_from_path.py
"""
Module for saving the metadata and the binary files from a data objects.

input: 
path_to_metadata - a string containing the path to the metadata file.

return: 
Return_list_data - a list of data objects loaded from the binary files described in the metadata.
Return_list_metadata - a list of metadata objects read from the metadata file.
Success - boolean value indicating if the data was loaded correctly.

"""

import os
import numpy as np
import json
import tsdf as tsdf


Success:bool = False

# path_to_metadata = "/Users/vedran/git/biomarkers_repo/tsdf4matlab/data/ppp_format_meta_legacy.json"

# Load the metadata
with open(path_to_metadata, "r") as file:
    metadata_dict = tsdf.load_metadata_file(file)

# List of metadata objects read from the metadata file
Return_list_metadata:list = []

# List of data objects loaded from the binary files
Return_list_data:list = []

# Load the data from each binary file described in the metadata
for metadata_dict_key in metadata_dict.keys():
    metadata = metadata_dict[metadata_dict_key]
    # Save the metadata
    Return_list_metadata.append(json.dumps(metadata.get_plain_tsdf_dict_copy()))
    # Save the data
    data = tsdf.load_binary_from_metadata(metadata)
    Return_list_data.append(data)

Success = True

print(len(Return_list_metadata))